### DISCUSSION TITLE
Cast storage not working as expected: Getting different output(null) while expecting the bytes output of myPassword

### DISCUSSION THREAD

**Discussion by skpkss** [Cyfrin/security-and-auditing-full-course-s23#230] [Q&A] [answered]
![q](https://github.com/user-attachments/assets/f66ee53d-e278-4fc2-aa27-97056c20e8e1)
Video lesson: https://updraft.cyfrin.io/courses/security/first-audit/proof-of-code?lesson_format=video
I am doing the password store audit. Currently, i am building on proof of code to show that private variable are not actually private. but when i am running the cast storage command it is displaying me the bytes of null instead of 'myPassword'

**Comment by EngrPips** [comment]
Hello @skpkss, I am convinced this is more a Foundry issue than the course issue. Please check the official Foundry GitHub to see if any similar issue has been reported recently.

**Comment by EngrPips** [comment]
You can also try running `foundryup` and then retry. Maybe that will help.

**Comment by skpkss** [comment]
I have already tried it.It's not helping.

**Reply by EngrPips** [reply]
I see. Did you find a recent report of a similar issue on the official GitHub page of Foundry?

**Comment by skpkss** [comment]
no. is this has to do something with the version of make? i am using 0.8.1

**Reply by EngrPips** [reply]
I think at a point in the past, this was an issue with Foundry. Around May Foundry update or so I think, but I haven't seen anyone run into this issue of recently.

**Comment by skpkss** [comment]
Can you run this at your end? My forge version is forge 0.2.0 (a0a0020 2024-08-12T00:25:05.209264900Z)

**Reply by EngrPips** [reply]
I have this 

```js
forge --version
forge 0.0.2 (1cf88dc 2024-08-21T00:21:46.971220000Z)
```

**Reply by EngrPips** [reply]
I ran `foundryup,` and now my versions are below.

```js
foundryup


.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

 ╔═╗ ╔═╗ ╦ ╦ ╔╗╔ ╔╦╗ ╦═╗ ╦ ╦         Portable and modular toolkit
 ╠╣  ║ ║ ║ ║ ║║║  ║║ ╠╦╝ ╚╦╝    for Ethereum Application Development
 ╚   ╚═╝ ╚═╝ ╝╚╝ ═╩╝ ╩╚═  ╩                 written in Rust.

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

Repo       : https://github.com/foundry-rs/
Book       : https://book.getfoundry.sh/
Chat       : https://t.me/foundry_rs/
Support    : https://t.me/foundry_support/
Contribute : https://github.com/orgs/foundry-rs/projects/2/

.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx.xOx

foundryup: installing foundry (version nightly, tag nightly)
foundryup: downloading latest forge, cast, anvil, and chisel
################################################################################################################################################################################# 100.0%
foundryup: downloading manpages
################################################################################################################################################################################# 100.0%
foundryup: installed - forge 0.2.0 (2b1f8d6 2024-08-26T00:23:04.542531000Z)
foundryup: installed - cast 0.2.0 (2b1f8d6 2024-08-26T00:23:01.593979000Z)
foundryup: installed - anvil 0.2.0 (2b1f8d6 2024-08-26T00:23:04.437933000Z)
foundryup: installed - chisel 0.2.0 (2b1f8d6 2024-08-26T00:23:03.094951000Z)
foundryup: done!

```

**Reply by skpkss** [reply]
i have the same foundry version.  Can you run the cast commands in this lesson: https://updraft.cyfrin.io/courses/security/first-audit/proof-of-code?lesson_format=transcript and please share the output?

**Comment by EngrPips** [comment]
So I did as you instructed here is my output 

**DEPLOYMENT**

```js
make deploy
[⠒] Compiling...
[⠔] Compiling 16 files with Solc 0.8.18
[⠒] Solc 0.8.18 finished in 2.42s
Compiler run successful!
EIP-3855 is not supported in one or more of the RPCs used.
Unsupported Chain IDs: 31337.
Contracts deployed with a Solidity version equal or higher than 0.8.20 might not work properly.
For more information, please see https://eips.ethereum.org/EIPS/eip-3855
Script ran successfully.

== Return ==
0: contract PasswordStore 0x5FbDB2315678afecb367f032d93F642f64180aa3

## Setting up 1 EVM.

==========================

Chain 31337

Estimated gas price: 2.000000001 gwei

Estimated total gas used for script: 450223

Estimated amount required: 0.000900446000450223 ETH

==========================

##### anvil-hardhat
✅  [Success]Hash: 0x77dceaade57b07bce8ee61e3047769709707dd11178e3e00deaccf59fe15032b
Contract Address: 0x5FbDB2315678afecb367f032d93F642f64180aa3
Block: 1
Paid: 0.000295344000295344 ETH (295344 gas * 1.000000001 gwei)


##### anvil-hardhat
✅  [Success]Hash: 0xc376757e1c78caf0e7c09709ac6231f1447098d3d784f06820a392c5a32d9fc8
Block: 2
Paid: 0.000039817434378978 ETH (45378 gas * 0.877461201 gwei)

✅ Sequence #1 on anvil-hardhat | Total Paid: 0.000335161434674322 ETH (340722 gas * avg 0.938730601 gwei)
                                                                                                                                                                       

==========================

ONCHAIN EXECUTION COMPLETE & SUCCESSFUL.

Transactions saved to: /Users/loyalty/Desktop/CyfrinGig/SECURITY&AUDITING/2023-10-PasswordStore/broadcast/DeployPasswordStore.s.sol/31337/run-latest.json

Sensitive values saved to: /Users/loyalty/Desktop/CyfrinGig/SECURITY&AUDITING/2023-10-PasswordStore/cache/DeployPasswordStore.s.sol/31337/run-latest.json

```

**RETRIEVING VALUE FROM STORAGE SLOT 1**

```js
cast storage 0x5fbdb2315678afecb367f032d93f642f64180aa3 1
0x6d7950617373776f726400000000000000000000000000000000000000000014

```
It seems to work properly for me.

**Reply by EngrPips** [reply]
Did you have an `Anvil` chain running in a separate terminal when you ran the `cast storage 0x5fbdb2315678afecb367f032d93f642f64180aa3 1` command?

**Reply by skpkss** [reply]
Yes, I am running anvil chain. I think there is issue on my side with `make deploy` command. I am getting this output when i'm running make deploy:  
```
$ make deploy
make i info Invoking deploy target
[⠘] Compiling...
No files changed, compilation skipped
Script ran successfully.
Gas used: 306748

== Return ==
0: contract PasswordStore 0x90193C961A926261B756D1E5bb255e67ff9498A1

If you wish to simulate on-chain transactions pass a RPC URL.
make i info Build success in 2229ms
```

My make version is 0.8.1 and i have installed it using `npm install make -g`, please check your version of make.

**Reply by EngrPips** [reply]
What is the command for the `make deploy` target in your `Makefile`?

**Reply by EngrPips** [reply]
Did I check make version appropriately?
```js
make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0

```

**Reply by skpkss** [reply]
can you share the make install commands? i think i have avery old version 0.8.1

**Reply by EngrPips** [reply]
Are you operating on a Mac?

**Reply by skpkss** [reply]
no on windows using git bash

**Reply by EngrPips** [reply]
Alright, Please read this `Ethereum stack exchange` post https://stackoverflow.com/questions/32127524/how-to-install-and-use-make-in-windows on how to install `make` for Windows.

**Reply by skpkss** [reply]
this is the make deploy command: 
```
deploy:
	@forge script script/DeployPasswordStore.s.sol:DeployPasswordStore $(NETWORK_ARGS)
```

**Reply by EngrPips** [reply]
I will search for a proper guide on how to install `make` for Windows and get back to you with whatever I find.

**Reply by skpkss** [reply, marked as answer]
Hey @EngrPips  the issue was resolved.First of all, i installed scoop from https://scoop.sh/ by running commands on powershell. And after that searched for make and ran the make installation commands on cmd.
![make](https://github.com/user-attachments/assets/2a6535a1-b0d1-4dd5-a7c7-8855d24eb414)

**Reply by EngrPips** [reply]
Interesting, Most of the time, our solution is not far-fetched, only if we can put in a little more effort! Keep crushing buddy
